modules/highlights.nix

```diff
@@ -35,50 +35,52 @@ with lib;
     };
   };
 
-  config = {
-    extraConfigLuaPre =
-      (optionalString (config.highlight != { })
-        # lua
-        ''
-          -- Highlight groups {{
-          do
-            local highlights = ${helpers.toLuaObject config.highlight}
+  config = mkMerge [
+    {
+      extraConfigLuaPre =
+        mkIf (config.highlight != { })
```

Comment:
Is using `mkIf` really so important that we must also use it when concatenating strings?

IMO concatenating optional strings is fine here and in similar cases, @traxys what do you think?

Reply:
There was two way to do it:

1) Concat strings in a variable, then in mkIf check if it empty.
2) Separate with mkMerge.

I can't decide what's best, I though that this way looks better than when two options are concated with variables and third as is. This way all three options looks the similar.

Reply:
Ah I see, so you do this because all parts of the string are optional, so you're avoiding an empty definition?

That makes more sense. Does that apply to all changes though?